gems/sorbet-runtime/lib/types/props/decorator.rb

```diff
@@ -510,8 +505,13 @@ def prop_defined(name, cls, rules={})
         end
       end
 
-      @class.send(:define_method, name) do
-        self.class.decorator.prop_get(self, name, rules)
+      if method(:prop_get).owner != T::Props::Decorator || rules.key?(:ifunset)
+        @class.send(:define_method, name) do
+          self.class.decorator.prop_get(self, name, rules)
+        end
+      else
+        # Fast path!
+        @class.attr_reader(name)
```

Comment:
Is it already the case that `prop_get` doesn't have runtime type checking for the result? Are we loosing that by switching to attr_reader?

Reply:
It's already the case.

(And TBH it doesn't worry me much, since it makes more sense to check at set time. The bigger issue IMHO is that while setters are checked, deserialization doesn't check types (only presence/absence).)